java/src/jmri/jmrit/display/palette/ItemPalette.java

```diff
@@ -347,22 +347,44 @@ static void loadMissingItemType(String itemType, Editor ed) {
         return familyTOMap;
     }
 
-    /**
-     * Build the Control Panel Editor tabbed Item Panel frame &amp; menus
-     */
-    public ItemPalette(String title, Editor editor) {
-        super(title, true, true);
-//        long t = System.currentTimeMillis();
-        loadIcons(editor);
+    static public ItemPalette getDefault(String title, Editor ed) {
+        if (GraphicsEnvironment.isHeadless()) {
+            return null;
+        }
+        ItemPalette instance = InstanceManager.getOptionalDefault(ItemPalette.class).orElseGet(() -> {
+            return InstanceManager.setDefault(ItemPalette.class, new ItemPalette(title, ed));
+        });
+        Iterator<Entry<String, ItemPanel>> iter = _tabIndex.entrySet().iterator();
+        if (ed != null) { // tests use null
```

Comment:
Is it valid for the editor to be null?

If it is, then this change is fine.

If it isn't, then the correct solution is to annotate the ed parameter as NonNull and then change the test(s) so they build an editor. Object to pass instead of null.